execution/gethexec/caff_node.go

```diff
@@ -189,9 +228,67 @@ func (n *CaffNode) Start(ctx context.Context) error {
 	if err != nil {
 		return fmt.Errorf("failed to start node, error in createBlock: %w", err)
 	}
+
+	err = n.CallIterativelySafe(func(ctx context.Context) time.Duration {
+		batch := n.db.NewBatch()
+		nextHotshotBlock, currentPosition := n.espressoStreamer.GetNextMessagePosAndHotShotBlock()
+		nextHotshotBytes, err := rlp.EncodeToBytes(nextHotshotBlock)
+		if err != nil {
+			log.Error("failed to encode next hotshot block", "err", err)
+		}
+		currentPositionBytes, err := rlp.EncodeToBytes(currentPosition)
+		if err != nil {
+			log.Error("failed to encode current position", "err", err)
+		}
+		err = batch.Put([]byte(NextHotshotBlockKey), nextHotshotBytes)
+		if err != nil {
+			log.Error("failed to put next hotshot block", "err", err)
+		}
+		err = batch.Put([]byte(CurrentPositionKey), currentPositionBytes)
+		if err != nil {
+			log.Error("failed to put current position", "err", err)
+		}
+
+		err = batch.Write()
+		if err != nil {
+			log.Error("failed to write batch", "err", err)
+		}
+
+		return time.Minute * 5
+	})
+	if err != nil {
+		return fmt.Errorf("failed to start node, error in persisting: %w", err)
+	}
 	return nil
 }
 
+func ReadCurrentPositionAndNextHotshotBlockFromDb(db ethdb.Database) (uint64, uint64, error) {
```

Comment:
Not needed, we should just update the database when the nextHotshotBlock number is updated